### PR TITLE
Feat/bubble chart

### PR DESCRIPTION
**Pull request by bas080**
- Changes to api
- Bubble chart feature
- New build

@RWOverdijk ready for a merge and a publish.

### CODE REVIEW

**Comment by VMBindraban** [comment]
Don't include the build into the PR.
Make a new PR for building and version bumping.

File: src/chart/two-dimensional.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ export class TwoDimensional extends C3Chart {
       const xLabel = this.dimensions[0].label(dataset);
       xs[label]    = xLabel;
       columns      = columns.concat([
-        [label].concat(this.dimensions[1].data(dataset.values)),
-        [xLabel].concat(this.dimensions[0].data(dataset.values))
+        [label ].concat(dataset.map(this.dimensions[1].value)),
```
Comment (VMBindraban) [inline src/chart/two-dimensional.js]: Remove space in the array.

File: src/chart/three-dimensional.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      x: this.dimensions[0].label(),
+      y: this.dimensions[1].label()
+    });
+
+    this.settings = {columns, xs};
+
+    return this.settings;
+  }
+
+
```
Comment (VMBindraban) [inline src/chart/three-dimensional.js]: Remove newlines

File: src/chart/one-dimensional.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ export class OneDimensional extends C3Chart {
     const columns = this.data.map(dataset => {
       const label = dataset.key;
 
-      return [label].concat(this.dimensions[0].data(dataset.values));
+      return [label].concat(dataset.map(this.dimensions[0].value));
```
Comment (VMBindraban) [inline src/chart/one-dimensional.js]: No need for `const`.

File: src/chart/bubble.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export class BubbleChart extends ThreeDimensional {
+
+  settings = {
+    grid: {
+      x: {show: true},
+      y: {show: true}
+    },
+    data: {
+      type: 'scatter',
+      columns: []
```
Comment (VMBindraban) [inline src/chart/bubble.js:17]: Alignment

**Review by VMBindraban — APPROVED** [review]

**Review by VMBindraban — CHANGES_REQUESTED** [review]
Pressed wrong button, see feedback.

File: src/chart/one-dimensional.js [hunk 5]
```diff
@@ -8,10 +8,10 @@ import {C3Chart} from './c3-chart';
 export class OneDimensional extends C3Chart {
 
   calculateSettings() {
-    const columns = this.data.map(dataset => {
-      const label = dataset.key;
+    let columns = this.data.map(dataset => {
+      let label = dataset.key;
 
-      return [label].concat(this.dimensions[0].data(dataset.values));
+      return [label].concat(dataset.map(this.dimensions[0].value));
```
Comment (VMBindraban) [inline src/chart/one-dimensional.js]: `return [dataset.key].concat(dataset.map(this.dimensions[0].value));`

File: src/chart/three-dimensional.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    let xs      = {};
+
+    this.dimensions.third = {};
+
+    this.data.forEach(dataset => {
+      const label  = this.dimensions[1].label(dataset);
+      const xLabel = this.dimensions[0].label(dataset);
+      xs[label]    = xLabel;
+      columns      = columns.concat([
+        [label ].concat(dataset.map(this.dimensions[1].value)),
```
Comment (VMBindraban) [inline src/chart/three-dimensional.js]: Remove space.

File: src/chart/three-dimensional.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    });
+
+    this.instance.axis.labels({
+      x: this.dimensions[0].label(),
+      y: this.dimensions[1].label()
+    });
+
+    this.settings = {columns, xs};
+
+    return this.settings;
```
Comment (VMBindraban) [inline src/chart/three-dimensional.js]: `return {columns, xs};`?

File: src/chart/two-dimensional.js [hunk 8]
```diff
@@ -15,10 +15,11 @@ export class TwoDimensional extends C3Chart {
     this.data.forEach(dataset => {
       const label  = this.dimensions[1].label(dataset);
       const xLabel = this.dimensions[0].label(dataset);
+
       xs[label]    = xLabel;
       columns      = columns.concat([
-        [label].concat(this.dimensions[1].data(dataset.values)),
-        [xLabel].concat(this.dimensions[0].data(dataset.values))
+        [label] .concat(dataset.map(this.dimensions[1].value)),
```
Comment (VMBindraban) [inline src/chart/two-dimensional.js]: Remove space before `.concat`

**Review by VMBindraban — CHANGES_REQUESTED** [review]

File: src/chart/two-dimensional.js (hunk 8 above)
Comment (bas080) [reply src/chart/two-dimensional.js]: Reason why I did this is to make it easier to edit and read.

File: src/chart/one-dimensional.js (hunk 5 above)
Comment (bas080) [reply src/chart/one-dimensional.js]: This has changed in a later version.

File: src/chart/bubble.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+ * radius of the scatter
+ */
+@scales([quan, quan, quan])
+@chart('C3', 'bubble')
+export class BubbleChart extends ThreeDimensional {
+
+  settings = {
+    data: {
+      type: 'scatter',
+      columns: []
```
Comment (VMBindraban) [inline src/chart/bubble.js:17]: Alignment

File: src/chart/bubble.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import {quan, scales, chart} from 'aurelia-charts';
+import {ThreeDimensional} from './three-dimensional';
```
Comment (VMBindraban) [inline src/chart/bubble.js:2]: Alignment

File: src/chart/three-dimensional.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import {C3Chart} from './c3-chart';
+import {TwoDimensional} from './two-dimensional';
```
Comment (VMBindraban) [inline src/chart/three-dimensional.js:1]: Alignment

File: src/chart/three-dimensional.js [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+ * All one dimensional charts share the same logic.
+ */
+export class ThreeDimensional extends C3Chart {
+
+  calculateSettings() {
+    /* only for two dimensional plots */
+
+    this.dimensions.third = {};
+
+    this.settings = TwoDimensional.prototype.calculateSettings.call(this);
```
Comment (VMBindraban) [inline src/chart/three-dimensional.js:16]: Alignment and remove newline

**Review by VMBindraban — CHANGES_REQUESTED** [review]

**Review by VMBindraban — APPROVED** [review]

**Comment by RWOverdijk** [comment]
@bas080 @VMBindraban was right. Don't include builds with PRs.

**Pull request merged**